pennylane/ops/qubit/parametric_ops.py

```diff
@@ -3028,3 +3028,144 @@ def adjoint(self):
 
     def pow(self, z):
         return [IsingXY(self.data[0] * z, wires=self.wires)]
+
+
+class PSWAP(Operation):
+    r"""Phase SWAP gate
+
+    .. math:: \mathtt{PSWAP}(\phi) = \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 0 & e^{i \phi} & 0 \\
+            0 & e^{i \phi} & 0 & 0 \\
+            0 & 0 & 0 & 1
+        \end{bmatrix}.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Gradient recipe:
+
+    .. math::
+        \frac{d}{d \phi} \mathtt{PSWAP}(\phi)
+        = \frac{1}{2} \left[ \mathtt{PSWAP}(\phi + \pi / 2) - \mathtt{PSWAP}(\phi - \pi / 2) \right]
+
+    Args:
+        phi (float): the phase angle
+        wires (int): the subsystem the gate acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+    """
+    num_wires = 2
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = "A"
+    grad_recipe = ([[0.5, 1, np.pi / 2], [-0.5, 1, -np.pi / 2]],)
```

Comment:
Was this required? Other qubit gates don't have it defined.

Reply:
yes, it is required for the parameter-shift diff method as there is no generator.